### DISCUSSION TITLE
Welcome to Poiji Discussions!

### DISCUSSION THREAD

**Discussion by ozlerhakan** [ozlerhakan/poiji#183] [General]
## 👋 Welcome to Poiji!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by bendh** [comment]
Are there any plans to create writer functionality? Would be nice if you could reuse annotation and have a simple writer functionality. Would like to contribute.

**Reply by ozlerhakan** [reply]
Hi @bendh , currently there is no plan for it. One of the contributors implemented it but the PR is close right now for his decision. You can do it and open up a PR however, be aware that this could take some time to merge since it will be a major update.

**Reply by bendh** [reply]
Thanks, Will look at it. Thanks for the release

**Comment by JayeshKumud** [comment]
Hi, First of all thank you for making such helpful utility,
I have come across situation where I have to generate List<Inner Class> from sheet, like below

@ExcelSheet("ModifyCustomerRequest")
public class ModifyCustomerRequest implements Request {

    @ExcelCellName("accNum")
    public String accNum;

    public List<DocumentDetail> documentDetails;

  public static class DocumentDetail {

        @ExcelCellName("documentType")
        public int documentType;

        @ExcelCellName("documentStatus")
        public int documentStatus;
    }
}

How would I achieve this, I have made separate sheet for DocumentDetail but still it doesn't work, Please help.

**Reply by JayeshKumud** [reply]
Try to add below ExcelSheet of inner class also but it did not work.

@ExcelSheet("ModifyCustomerRequest")
public class ModifyCustomerRequest implements Request {

    @ExcelCellName("accNum")
    public String accNum;

    public List<DocumentDetail> documentDetails;

   @ExcelSheet("DocumentDetail")
    public static class DocumentDetail {

        @ExcelCellName("documentType")
        public int documentType;

        @ExcelCellName("documentStatus")
        public int documentStatus;
    }
}

**Reply by ozlerhakan** [reply]
Hey @JayeshKumud ,

Thank you for using Poiji, first of :) 

The mentioned approach `public List<DocumentDetail> documentDetails;` doest not work in Poiji :( 

What It works is that List supports primitive types like `public List<Integer> integers;` and the corresponding cell must be "1;2;3;4;5".

We can have this option If you are willing to add it via PR :)

**Reply by JayeshKumud** [reply]
@ozlerhakan , As you mention I have added it via PR "New feature requested to have List<Inner class> in Parent pojo class #191", can you please have a look?

**Reply by ozlerhakan** [reply]
Hi @JayeshKumud , probably I did not explain clearly to you. What I actually meant that since we know that this is not possible with the currenct Poiji, we need to implement this feature in order to make it possible. So about the PR you opened, I expect the implementation of this feature, no need a PR to request a feature 🤗

**Reply by ozlerhakan** [reply]
I am closing it right now and feel free to implement this behavior and open a PR about the changes to use in Poiji 😎

**Reply by ozlerhakan** [reply]
did you find a way to implement it @JayeshKumud ?

**Comment by oradchenko1** [comment]
Hi! Thanks a lot for such a nice library! It helped a lot. But now I have a situation when I need to deserialize huge xlsx file ~90Mb.
And I have issues with performance - almost 4GB of RAM is consumed every time I call Poiji with this file. 
Will there be any option to stream the content of xlsx somehow?

**Reply by oradchenko1** [reply]
I found .skip and .limit in OptionsBuilder.. Wonder if I could use it as a pagination. Also would be interesting to know if there any possibility to make async read in thread pool of xlsx rows.. for example 1 thread reads rows from 1 to 1000, another one from 1001 to 2000 etc.

**Reply by ozlerhakan** [reply]
Hi @oradchenko1 , thank you for using Poiji :) currently It doesn't provide a stream api, we can have one by implementing it though.  Plus, we don't have a sync feature as well right now. If you want you can always contribute the library for these features as well :)
